_posts/2021-09-21-release-1.14.0.md

```diff
@@ -0,0 +1,274 @@
+  <img src="{{ site.baseurl }}/img/blog/2021-09-25-release-1.14.0/buffer_debloating.png" style="display: block; margin-left: auto; margin-right: auto; width: 600px"/>
+</figure>
+
+Buffer Deloating acts as a complementary feature, or even alternative, to unaligned checkpoints.
+Checkout [the docs](https://nightlies.apache.org/flink/flink-docs-release-1.14/docs/deployment/memory/network_mem_tuning/#the-buffer-debloating-mechanism)
+to see how to activate this feature.
+
+
+## Fine-grained Resource Management
+
+The new *Fine-grained Resource Management* is an advanced feature to increase the resource
+utilization of large shared clusters.
+
+Flink clusters execute various data processing workloads, and different data processing steps typically need
+different resources: compute resource, memory, etc. For example a filter in SQL needs different
+resources than a join. By default, Flink manages resources in coarse-grained
+units called 'slots', which are slices of a TaskManager's resources. A slot is then filled with tasks
+that use those resources. Through *'slot sharing groups'*, users can influence how the tasks are put
+into slots.
+
+With fine-grained resource management, TaskManager slots are now dynamically sized. Transformations and Operators
+specify what resource profiles they would like, and Flink's Resource Manager and TaskManagers slice
+off that specific part of a TaskManagers total resources. Think of it as a minimal lightweight resource
+orchestration layer within Flink. Hmmmm, why would we even do that, is this not better handled directly by
+Kubernetes or Yarn? For most cases, that is in fact true. But we have seen that in large shared clusters,
+resource requirements change so fast (especially in SQL jobs where each operator needs different resources),
+that a lot of efficiency gets lost waiting for Yarn and K8s to fullfill the resource requests (taking seconds,
+where Flink internally does it in milliseconds).
+
+This feature is mainly useful for large platform deployments where Flink runs many batch
+jobs of varying kind and manages the resources actively (dynamically requesting and releasing resources
+from Kubernetes or Yarn, as opposted to opaque reactive deployments). Please refer to the
+[Fine-grained Resource Management Docs](https://nightlies.apache.org/flink/flink-docs-release-1.14/docs/deployment/finegrained_resource/)
+for details on how to use this feature.
+
+<figure style="align-content: center">
+  <img src="{{ site.baseurl }}/img/blog/2021-09-25-release-1.14.0/fine_grained_resource_management.png" style="display: block; margin-left: auto; margin-right: auto; width: 600px"/>
+</figure>
+
+Alibaba's internal SQL platform on top of Flink and has used this mechanism for some years now and
+has seen significantly increase the resource utilization of the cluster. This feature was open sourced
+to help other users that build large platforms on top of Flink to optimize resource footprints.
+
```

Comment:
1. This feature serves both long running streaming jobs and short lived batch / bounded-stream jobs. IMHO, it's is more helpful to the long running jobs than to the batch jobs. Short lived jobs tend to prioritize scheduling overhead over resource utilization.

2. Why not to leverage the orchestration of K8s / Yarn directory?
- Avoid waiting for pod / container allocation and initialization (as mentioned) is one of the reasons.
- Slots are sometimes small. Leveraging multi-slot TMs is more efficient than requesting many TMs, by sharing the framework overhead. Moreover, having too many small pods / containers also create significant tensions for K8s / Yarn.
- The key of fine-grained resource management is to have dynamic sized slots. Slicing these slots from TMs with the same predefined size is only the first step, with minimum changes to the previous mechanism. A potential next step would be having dynamic sized TMs, where the sizes are derived from the requested slots. In other words, having dynamic sized slots / TMs is a precondition for better leveraging orchestration of K8s / Yarn.

3. I'd suggest not to emphasize the SQL use cases. This feature is only available to DataStream jobs in Flink 1.14.

Reply:
This figure might help demonstrate the benefit of this feature, comparing with coarse-grained resource management.
https://drive.google.com/file/d/1-xpwluhzNy_N44a5Z0OOygIduQgkfWR9/view?usp=sharing

Reply:
I adjusted the text and used your new figure (removed the non-slot-sharing part, to create less confusion).